components/responsive.vue

```diff
@@ -130,6 +131,12 @@ export default
 			{ provider, preset } = @$props
 			@$cloakSrcset source, {}, { provider, preset }
 
+		# Add srcset to customResponsiveSources
+		addSrcsetToCustomSources: (sources) ->
+			sources[0]?.attrs?.srcset = @makeSrcset @landscapeImage
+			sources[1]?.attrs?.srcset = @makeSrcset @portraitImage
+			return sources
```

Comment:
I think this is too specific to your current use case.  Like, if someone is providing `customResponsiveSources`, there may be more than 2 `sources` in the array and the first `source` may not use the `@landscapeImage`.  

If you want to use the `makeSrcset` method, then what I'd recommend is updating the `customResponsiveSources` to also support a Function prop.  Then you'd need to test for what type of prop was provided.  If a function, you would invoke it, binding it to the `this` scope of the component, so you could use  `makeSrcset`.  Like:

```coffee
responsiveSources: ->
  if @customResponsiveSources
    if typeof @customResponsiveSources == 'function'
    then return @customResponsiveSources.call(this)
    else return @customResponsiveSources
```

Reply:
@weotch 
I understand your logic, though I have tried and it seemed to bug because the component methods would execute first before parsing as props to `responsive-craft-visual` and at that time, it could not recognize `@makeSrcset` yet.

Reply:
Can you push a PR with your Essentials changes and I'll see if I can get this to work?

Reply:
@weotch Let's not push this furthur, I damnly figure out that even if we enforce `srcset` - iframe still put `reponsive-visual` to recognize as `orientation: portrait` and misbehave the stylings. I will use two individual single image fields instead :(

Reply:
Ok, kk,  gonna close this PR then